localstack/utils/aws/aws_stack.py

```diff
@@ -1098,7 +1098,14 @@ def apigateway_invocations_arn(lambda_uri):
     )
 
 
-def get_elasticsearch_endpoint(region_name: str, domain_arn: str = None):
+def get_elasticsearch_endpoint(region_name: str, domain_arn: Optional[str] = None) -> str:
+    """
+    Get an ElasticSearch cluster endpoint by describing the cluster associated with the domain_arn
+    :param region_name: cluster region e.g. us-east-1
+    :param domain_arn: ARN of the cluster. Uses TEST_ELASTICSEARCH_URL env var if not set.
+    :returns: cluster endpoint
+    """
+    # TODO: extract region_name from the domain_arn
     if not domain_arn:
         return os.environ["TEST_ELASTICSEARCH_URL"]
```

Comment:
It's not clear to me if this is necessary or how it's intended to be used. I couldn't find any other references to the env var name. Can it be removed?

Reply:
Good catch! Actually, due to changes on the routing for opensearch / elasticsearch, this doesn't make sense anymore.
The proper handling would be:
- Only take the domain ARN (non-optional) as a parameter.
- Extract the domain name and the region from the ARN.
  - You can use `aws_stack.py#extract_region_from_arn` for the region. The domain name parsing can stay the same.
- Use the elasticsearch / opensearch API to get the domain info with the correct endpoint (as it's already done in the lines below).

This function is only used by `connect_elasticsearch` (which in turn is only used by Firehose), so it should be fairly straight-forward / safe to change.
Since the function is not used anywhere outside this file, it is safe to just change it.

Reply:
Also, since the APIs are interconnected with https://github.com/localstack/localstack/pull/5341, you could also rename this function to `get_opensearch_endpoint` and remove the new one below (opensearch has priority over elasticsearch from now on).

Reply:
very cool!